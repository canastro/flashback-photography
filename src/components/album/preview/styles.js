import {scale, rhythm} from '../../../utils/typography';
import presets from '../../../utils/presets';

export default {
    root: {
        display: 'block',
        backgroundColor: 'lightgray',
        flex: '1 0 0%',
        marginRight: rhythm(1 / 8),
        width: '100%',
        maxWidth: 290.1,
        position: 'relative',
        [presets.Tablet]: {
            marginRight: rhythm(1)
        },
        ':last-child': {
            marginRight: 0
        },
        color: 'inherit',
        textTransform: 'uppercase',
        textDecoration: 'none'
    },
    wrapper: {
        flexDirection: 'column',
        flexShrink: 0,
        position: 'relative',
        overflow: 'hidden',
        outline: 0
    },
    image: {
        margin: 0,
        height: '100%',
        width: '100%',
        verticalAlign: 'baseline',
        position: 'absolute',
        top: 0,
        bottom: 0,
        left: 0,
        right: 0,
        outline: 'none'
    },
    overlay: {
        position: 'absolute',
        top: 0,
        bottom: 0,
        left: 0,
        right: 0,
        backgroundColor: 'rgba(0,0,0,0.3)',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        color: 'white',
        ...scale(2 / 5)
    },
    icon: {
        fontSize: '90%',
        marginRight: rhythm(1 / 4)
    },
    title: {
        position: 'absolute',
        bottom: 0,
        height: 40,
        backgroundColor: 'white',
        width: '100%',
        opacity: 0.8,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center'
    }
};
