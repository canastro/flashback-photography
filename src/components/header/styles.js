import {rhythm, scale} from '../../utils/typography';

export const burgerStyles = {
    bmBurgerButton: {
        position: 'absolute',
        width: '36px',
        height: '30px',
        left: 'initial',
        top: 20,
        right: 36
    },
    bmBurgerBars: {
        background: '#373a47'
    },
    bmCrossButton: {
        height: '24px',
        width: '24px'
    },
    bmCross: {
        background: '#bdc3c7'
    },
    bmMenuWrap: {
        top: 0
    },
    bmMenu: {
        background: '#373a47',
        padding: '2.5em 1.5em 0',
        fontSize: '1.15em'
    },
    bmMorphShape: {
        fill: '#373a47'
    },
    bmItemList: {
        color: '#b8b7ad',
        padding: '0.8em',
        display: 'flex',
        flexDirection: 'column'
    },
    bmOverlay: {
        background: 'rgba(0, 0, 0, 0.3)',
        left: 0,
        top: 0
    }
};

export default {
    root: {
        background: 'white',
        borderBottom: '1px solid rgba(0,0,0,0.08)',
        position: 'fixed',
        top: 0,
        left: 0,
        right: 0,
        zIndex: 1
    },
    wrapper: {
        maxWidth: 960,
        margin: '0 auto',
        overflow: 'hidden',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-between'
    },
    titleLink: {
        padding: rhythm(3 / 4),
        display: 'inline-block',
        float: 'left',
        textDecoration: 'none'
    },
    title: {
        ...scale(4 / 5),
        lineHeight: 1,
        margin: 0,
        overflow: 'hidden'
    },
    titleSpan: {
        paddingLeft: `calc(${rhythm(1)} - 1px)`,
        borderLeft: '1px solid rgba(0,0,0,0.3)',
        lineHeight: 1,
        marginLeft: rhythm(1)
    },
    icon: {
        top: -4,
        display: 'inline-block',
        position: 'relative'
    },
    link: {
        color: 'inherit',
        display: 'inline-block',
        float: 'right',
        lineHeight: '35px',
        textDecoration: 'none',
        marginLeft: 20,
        '&:hover': {
            color: '#000'
        }
    }
};
