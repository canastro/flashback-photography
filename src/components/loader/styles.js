export default {
    root: {
        position: 'fixed',
        height: '100%',
        width: '100%',
        top: 0,
        left: 0,
        backgroundColor: 'black',
        opacity: 0.5,
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        zIndex: 10
    },
    indicator: {
        display: 'inline-block',
        color: '#fff',
        width: 100
    },
    svgIndeterminate: {
        animation: 'circular-rotate 1.4s linear infinite'
    },
    circle: {
        stroke: 'currentColor',
        strokeLinecap: 'round',
        animation: 'circular-dash 1.4s ease-in-out infinite',
        // Some default value that looks fine waiting for the animation to kicks in.
        strokeDasharray: '80,200',
        strokeDashoffset: 0
    },
    '@keyframes circular-rotate': {
        '100%': {
            transform: 'rotate(360deg)'
        }
    },
    '@keyframes circular-dash': {
        '0%': {
            strokeDasharray: '1,200',
            strokeDashoffset: 0
        },
        '50%': {
            strokeDasharray: '100,200',
            strokeDashoffset: -15
        },
        '100%': {
            strokeDasharray: '100,200',
            strokeDashoffset: -120
        }
    }
};
