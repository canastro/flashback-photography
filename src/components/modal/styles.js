import {rhythm} from '../../utils/typography';

export default {
    wrapper: {
        display: 'flex',
        position: 'relative',
        height: '100vh'
    },
    caretsWrapper: {
        display: 'flex',
        maxWidth: rhythm(40.25),
        width: '100%',
        marginTop: 20,
        marginLeft: 'auto',
        marginRight: 'auto'
    },
    caretLeft: {
        cursor: 'pointer',
        fontSize: '50px',
        color: 'rgba(255,255,255,0.7)',
        userSelect: 'none',
        marginTop: 215
    },
    caretRight: {
        cursor: 'pointer',
        fontSize: '50px',
        color: 'rgba(255,255,255,0.7)',
        userSelect: 'none',
        marginTop: 215
    },
    close: {
        cursor: 'pointer',
        color: 'rgba(255,255,255,0.8)',
        fontSize: '30px',
        position: 'absolute',
        top: rhythm(1 / 4),
        right: rhythm(1 / 4)
    }
};
