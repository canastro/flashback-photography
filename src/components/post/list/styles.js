import {rhythm, scale} from '../../../utils/typography';
import presets from '../../../utils/presets';

export default {
    root: {
        display: 'flex',
        alignItems: 'stretch',
        flexShrink: 0,
        flexDirection: 'column'
    },
    postsChunk: {
        display: 'flex',
        alignItems: 'stretch',
        flexShrink: 0,
        flexDirection: 'row',
        marginBottom: rhythm(1 / 8),
        [presets.Tablet]: {
            marginBottom: rhythm(1)
        }
    },
    showMore: {
        ...scale(-0.5),
        border: '1px solid black',
        boxShadow: 0,
        background: 'none',
        color: 'black',
        cursor: 'pointer',
        margin: '0 auto',
        padding: rhythm(1 / 2),
        width: `calc(100vw - ${rhythm(1)})`,
        marginLeft: rhythm(0.5),
        marginRight: rhythm(0.5),
        marginBottom: rhythm(0.5),
        marginTop: rhythm(0.5),
        [presets.Tablet]: {
            margin: '0 auto',
            marginBottom: rhythm(1.5),
            marginTop: rhythm(1.5),
            height: rhythm(2),
            width: rhythm(5),
            textAlign: 'center'
        },
        '&:hover': {
            backgroundColor: '#ccc'
        }
    }
};
