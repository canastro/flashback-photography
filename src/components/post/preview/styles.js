import {scale, rhythm} from '../../../utils/typography';
import presets from '../../../utils/presets';

export default {
    root: {
        display: 'block',
        backgroundColor: 'black',
        flex: '1 0 0%',
        marginRight: rhythm(1 / 8),
        width: '100%',
        maxWidth: 290.1,
        position: 'relative',
        [presets.Tablet]: {
            marginRight: rhythm(1)
        },
        ':last-child': {
            marginRight: 0
        }
    },
    wrapper: {
        width: '100%',
        height: '100%',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center'
    },
    image: {
        maxHeight: 200
    },
    overlay: {
        position: 'absolute',
        top: 0,
        bottom: 0,
        left: 0,
        right: 0,
        backgroundColor: 'rgba(0,0,0,0.3)',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        color: 'white',
        ...scale(2 / 5)
    },
    icon: {
        fontSize: '90%',
        marginRight: rhythm(1 / 4)
    }
};
