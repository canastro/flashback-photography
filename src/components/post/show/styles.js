import presets from '../../../utils/presets';
import typography, {rhythm, scale} from '../../../utils/typography';

export default {
    root: {
        maxHeight: 500,
        backgroundColor: 'black',
        display: 'flex',
        alignItems: 'stretch',
        flexDirection: 'column',
        width: '100%',
        [presets.Tablet]: {
            flexDirection: 'row-reverse'
        }
    },
    detailsContainer: {
        ...scale(-2 / 5),
        backgroundColor: 'white',
        lineHeight: typography.options.baseLineHeight,
        height: '100%'
    },
    detailsContent: {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'space-around',
        height: 'calc(100% - 80px)'
    },
    date: {
        color: 'rgba(0,0,0,0.4)',
        float: 'right'
    },
    tabletDetailsContainer: {
        backgroundColor: 'white',
        padding: rhythm(3 / 4),
        paddingBottom: 0,
        [presets.Tablet]: {
            width: rhythm(13),
            padding: rhythm(1)
        }
    },
    tabletDetailsWrapper: {
        display: 'none',
        [presets.Tablet]: {
            display: 'block',
            height: '100%'
        }
    },
    mobileDetailsWrapper: {
        background: 'white',
        padding: rhythm(3 / 4),
        display: 'block',
        [presets.Tablet]: {
            display: 'none'
        }
    },
    imageContainer: {
        backgroundColor: 'lightgray',
        flex: '1 0 0',
        margin: 'auto',
        width: '100%'
    },
    listContainer: {
        display: 'flex',
        flexDirection: 'column'
    },
    settingItem: {
        listStyle: 'none',
        padding: 0,
        height: 25,
        display: 'flex',
        alignItems: 'center'
    },
    icon: {
        marginRight: 10
    },
    tag: {
        backgroundColor: '#ccc',
        margin: '10px 0 0 0',
        textAlign: 'center',
        width: 100,
        color: 'inherit',
        textDecoration: 'none',
        '&:hover': {
            backgroundColor: '#696969'
        }
    }
};
