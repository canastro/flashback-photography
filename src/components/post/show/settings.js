// @flow
import React from 'react';
import injectSheet from 'react-jss';

import timer from '../../../../static/images/timer.png';
import aperture from '../../../../static/images/aperture.png';
import camera from '../../../../static/images/camera.png';
import focalLength from '../../../../static/images/focal-length.png';
import iso from '../../../../static/images/iso.png';

import styles from './styles';

type Props = {
    classes: Object,
    settings: ?Object
};

/**
 * Photo settings, camera configurations and conditions
 * @method Settings
 * @param  {Object} props - react props
 * @returns {Node} react node
 */
const Settings = (props: Props) => {
    if (!props.settings) return null;

    return (
        <ul className={props.classes.listContainer}>
            <li className={props.classes.settingItem}>
                <img className={props.classes.icon} alt="Machine" src={camera} />
                {props.settings.machine}
            </li>
            <li className={props.classes.settingItem}>
                <img className={props.classes.icon} alt="Focal Length" src={focalLength} />
                {props.settings.focalLength}
            </li>
            <li className={props.classes.settingItem}>
                <img className={props.classes.icon} alt="Aperture" src={aperture} />
                {props.settings.aperture}
            </li>
            <li className={props.classes.settingItem}>
                <img className={props.classes.icon} alt="Exposure Time" src={timer} />
                {props.settings.exposureTime}
            </li>
            <li className={props.classes.settingItem}>
                <img className={props.classes.icon} alt="ISO" src={iso} />
                {props.settings.iso}
            </li>
        </ul>
    );
};

export default injectSheet(styles)(Settings);
