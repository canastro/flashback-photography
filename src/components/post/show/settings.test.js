import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import {Settings} from './settings';
import styles from './styles';
import {getClasses} from '../../../utils/tests';


const classes = getClasses(styles);

describe('when no settings are provided', () => {
    it('should render correctly', () => {
        const output = shallow(<Settings classes={classes} />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});


describe('when settings are provided', () => {
    it('should render correctly', () => {
        const settings = {
            iso: 'TEST-ISO',
            exposureTime: 'TEST-EXPOSURE-TIME',
            aperture: 'TEST-APERTURE',
            focalLength: 'TEST-FOCAL-LENGTH',
            machine: 'TEST-MACHINE'
        };

        const output = shallow(<Settings classes={classes} settings={settings} />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
