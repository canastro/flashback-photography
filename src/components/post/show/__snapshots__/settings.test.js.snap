// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when no settings are provided should render correctly 1`] = `""`;

exports[`when settings are provided should render correctly 1`] = `
<ul
  className="listContainer"
>
  <li
    className="settingItem"
  >
    <img
      alt="Machine"
      className="icon"
      src={null}
    />
    TEST-MACHINE
  </li>
  <li
    className="settingItem"
  >
    <img
      alt="Focal Length"
      className="icon"
      src={null}
    />
    TEST-FOCAL-LENGTH
  </li>
  <li
    className="settingItem"
  >
    <img
      alt="Aperture"
      className="icon"
      src={null}
    />
    TEST-APERTURE
  </li>
  <li
    className="settingItem"
  >
    <img
      alt="Exposure Time"
      className="icon"
      src={null}
    />
    TEST-EXPOSURE-TIME
  </li>
  <li
    className="settingItem"
  >
    <img
      alt="ISO"
      className="icon"
      src={null}
    />
    TEST-ISO
  </li>
</ul>
`;
